libraries/testing/tester.cpp

```diff
@@ -322,6 +323,34 @@ namespace eosio::testing {
       open( make_protocol_feature_set(), expected_chain_id );
    }
 
+   bool base_tester::_check_signal(const block_id_type& id, block_signal sig) {
+      auto update_sig = fc::make_scoped_exit([&]() { blocks_signaled[id] = sig; });
+
+      auto itr = blocks_signaled.find(id);
+      bool present = itr != blocks_signaled.end();
+
+      switch(sig) {
+      case block_signal::block_start:
+         return true;  // only block number is signaled,
+
+      case block_signal::accepted_block_header:
+         // should get accepted_block_header signal only once, and before accepted_block signal;
+         return !present;
+
+      case block_signal::accepted_block:
+         // should get accepted_block signal after accepted_block_header signal
+         // or after accepted_block (on fork switch, accepted block signaled when block re-applied)
+         return present && (itr->second == block_signal::accepted_block_header ||
+                            itr->second == block_signal::accepted_block);
+
+      case block_signal::irreversible_block:
+         return !present ||
```

Comment:
restore comment I forgot in my suggestion
```suggestion
         // can be signaled on restart as the first thing since other signals happened before shutdown
         // otherwise should get irreversible_block signal only once
         return !present ||
```